include/libcyphal/transport/can/transport.hpp

```diff
@@ -282,11 +298,18 @@ class TransportImpl final : public ICanTransport, private TransportDelegate  //
 
     // MARK: IRunnable
 
-    void run(const TimePoint now) override
+    CETL_NODISCARD IRunnable::MaybeError run(const TimePoint now) override
     {
         runMediaTransmit(now);
         runMediaReceive();
-        runMediaFilters();
+
+        cetl::optional<AnyError> any_error = runMediaFilters();
+        if (any_error.has_value())
+        {
+            return any_error.value();
+        }
```

Comment:
I don't get this part though; shouldn't we return `AnyErrorReport` here to match the transient error handler?

Reply:
I believe that result of generic `run` (which is `MaybeError`) shouldn't have such details - this is transport layer handler what for (with it's knowledge what *network* layer `culprit` could be). Yes, we probably could embed `AnyErrorReport` into `MaybeError`, but I'm very worrying about amount of indirection, and furthermore that higher level code needs to know about lower level "transport" stuff (like CAN specific `transport::can::IMedia*` as a culprit). Here is an example (even without `nullptr` checking):
```
auto maybe_platform_err = cetl::get_if<transport::IPlatformError>(cetl::get_if<transport::AnyErrorReport>(&me)→error));
auto maybe_can_media = get_if<transport::can::IMedia*>IPlatformError>(
    *get_if<transport::AnyError>(get_if<AnyErrorReport>(&me)));
```
and `maybe_can_media` here is actually double pointer like `transport::can::IMedia**`.

Reply:
Also, notice that I embedded `AnyErrorReport` into `transport::ITransport` - the idea is that this report is only some transport concern (with its expected **internal** knowledge about its possible media or whatever **its level** entity culprit might be).

Reply:
![image](https://github.com/OpenCyphal-Garage/libcyphal/assets/3298404/b79fcad8-797b-4a59-93c9-7025623fc0c1)

Reply:
>but probably we can write a visit function for unbounded_variant

If we do that, it would be rather different compared to the variant visitor because we don't know the set of possible types at compile time. We could supply the set of type alternatives at the time of invoking the visitor itself and provide a fallback case when there's no match but it's probably not going to be super easy to use.

Reply:
Yup, that is exactly what I have in mind.